ofrak_core/ofrak/core/elf/analyzer.py

```diff
@@ -333,9 +336,26 @@ async def analyze(self, resource: Resource, config=None) -> ElfSection:
         except ValueError:
             LOGGER.info("String section is empty! Using '<no-strings>' as section name")
             section_name = "<no-strings>"  # This is what readelf returns in this situation
-        return ElfSection(
-            section_index=section_header.section_index,
+        return AttributesType[NamedProgramSection](
             name=section_name,
+        )
+
+
+class ElfSectionMemoryRegionAnalyzer(Analyzer[None, MemoryRegion]):
+    """
+    Get the name of an ELF section. ELF section names are stored as null-terminated strings in
+    dedicated string section, and each ELF section header's `sh_name` field is an offset in this
+    section.
```

Comment:
I think this docstring is incorrect – it's supposed to be for the other analyzer, right?